src/pages/home/HeaderView.js

```diff
@@ -50,69 +51,124 @@ const defaultProps = {
     report: null,
 };
 
-const HeaderView = (props) => {
-    const participants = lodashGet(props.report, 'participants', []);
-    const reportOption = {
-        text: lodashGet(props.report, 'reportName', ''),
-        tooltipText: getReportParticipantsTitle(participants),
-        participantsList: getPersonalDetailsForLogins(participants, props.personalDetails),
-    };
-
-    return (
-        <View style={[styles.appContentHeader]} nativeID="drag-area">
-            <View style={[styles.appContentHeaderTitle, !props.isSmallScreenWidth && styles.pl5]}>
-                {props.isSmallScreenWidth && (
-                    <Pressable
-                        onPress={props.onNavigationMenuButtonClicked}
-                        style={[styles.LHNToggle]}
-                    >
-                        <Icon src={BackArrow} />
-                    </Pressable>
-                )}
-                {props.report && props.report.reportName && (
-                    <View
-                        style={[
-                            styles.flex1,
-                            styles.flexRow,
-                            styles.alignItemsCenter,
-                            styles.justifyContentBetween,
-                        ]}
-                    >
+class HeaderView extends Component {
+    constructor(props) {
+        super(props);
+
+        this.participants = lodashGet(this.props.report, 'participants', []);
+        this.reportOption = {
+            text: lodashGet(this.props.report, 'reportName', ''),
+            tooltipText: getReportParticipantsTitle(this.participants),
+            participantsList: getPersonalDetailsForLogins(this.participants, this.props.personalDetails),
+        };
+        this.toggleVideoChatMenu = this.toggleVideoChatMenu.bind(this);
+        this.measureVideoChatIconPosition = this.measureVideoChatIconPosition.bind(this);
+        this.videoChatIconWrapper = null;
+
+        this.state = {
+            isVideoChatMenuActive: false,
+            videoChatIconPosition: {x: 0, y: 0},
+        };
+    }
+
+    toggleVideoChatMenu() {
+        this.setState(prevState => ({
+            isVideoChatMenuActive: !prevState.isVideoChatMenuActive,
+        }));
+    }
+
+    measureVideoChatIconPosition() {
```

Comment:
Needs a doc

Reply:
I wonder if we can set up a linting rule for this. Not sure how much milage out of that we'd get, but just a thought.